### PR TITLE
Tweak .codedoc.yml to make it more useful

### PR DESCRIPTION
**Pull request by berkerpeksag**

### CODE REVIEW

File: .codecov.yml
Comment (briancurtin) [inline .codecov.yml]: I'm guessing this means it only ends up being run for those directories, right? If so, that's good that it avoids being run on, e.g., `Doc/*` changes, but shouldn't it also run on directories like `Objects/`?

Comment (berkerpeksag) [reply .codecov.yml]: > I'm guessing this means it only ends up being run for those directories, right?

Correct.

> [...]  but shouldn't it also run on directories like Objects/?

Good point. I'm reading docs again so I hopefully will have an answer  :)

Comment (berkerpeksag) [reply .codecov.yml]: According to https://docs.codecov.io/docs/commit-status#section-patch-status, we don't need to add specific directories. It should only look for the changes in PR.

We can also add directories to `ignore`: https://docs.codecov.io/docs/ignoring-paths

**Review by briancurtin — COMMENTED** [review]
This seems ok from what I understand of the docs (particularly the coverage/status/{changes|project} part), but I'd prefer to leave it to another set of eyes to approve.

File: .codecov.yml
Comment (brettcannon) [inline .codecov.yml:11]: Probably should add the Grammar directory.

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
LGTM. If you can pass it through Codecov's verification tool at https://docs.codecov.io/docs/codecov-yaml#section-validate-your-repository-yaml that would be great.

File: .codecov.yml
Comment (berkerpeksag) [reply .codecov.yml:11]: Good catch, done.

**Comment by berkerpeksag** [comment]
Thanks for the review!

> If you can pass it through Codecov's verification tool at https://docs.codecov.io/docs/codecov-yaml#section-validate-your-repository-yaml that would be great.

```sh
$ curl -XPOST --data-binary @.codecov.yml https://codecov.io/validate
Valid!

# Pretty-printed version of the configuration below
```

**Comment by brettcannon** [comment]
Then I say merge it and let python-dev (or -committers; wherever @haypo emailed) know about the requisite change.

**Pull request merged**
